app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt

```diff
@@ -578,4 +578,12 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
         skillDetailsPresenter.onDetach()
         super.onDestroyView()
     }
+
+    override fun onResume() {
+        if (PrefManager.getBoolean(R.string.is_feedback_deleted, false)) {
```

Comment:
Why?

Reply:
If a user deletes his or her feedback in the allReviewsActivity, then also he is able to do so. It is deleted in the server. Also, we remove it from the local list that contains the list of the data. Now, if user press backs, his feedback would be shown. This is because we are not fetching the data again. And in the allReviewsActivity we deleted in the variable of that file along with the server. It has not been deleted in the skillDetailsFragment. So, I used this preference to know if the user has deleted his feedback in allReviewActivity or not. If he has done so, then we are reloading the feedbacks only once again. Else, we are not making any API calls.

Reply:
Terrible solution. @liveHarshit Please tell what is done in normal apps where we delete something and need to show updated list

Reply:
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/43731599/60381730-0d833880-9a77-11e9-943d-47afe7161da6.gif)

Let me explain to u in detail why I did so.
Clearly, in the video, u can see there is an option "See All Reviews", pressing it redirects to a new page "AllReviewsActivity". In the intent, the feedbackList is passed as an extra to the intent. Now, there might be a situation as I showed u the gif that users delete his feedback while he/she is in "AllReviewsActivity". When he does so, we are also deleting it from the local list of the page also in the server. Now, the user will surely press back-button. If, he does so, the
"SkillDetailsFragment" is popped back. We don't refresh any data if the user does show. Now, as the user has already deleted his feedback, it should not appear back in the "SKillDetailsFragment". Now, the question arises who to know whether the user has deleted the feedback while he was in "AllReviewsActivity". To know that I used the preference manager. If the user has deleted the item we have 2 options to show the changes to the user. One is to delete the item from the feedback list and show it to the user. The other one is to reload the feedback list back from the server. In the second option, only the feedback list is reloaded not the other data of the page. A similar approach was already in use in the app, where if the feedback is added/updated by the user the feedback list is reloaded.

@iamareebjamal @liveHarshit please provide your views now.

Reply:
> Now, the question arises *how* to know whether the user has deleted the feedback while he was in "AllReviewsActivity"

Very simple, it's deleted from the database so it won't be there

Reply:
We don't have any database setup for storing the feedbacks. It's simply stored into variables.

Reply:
Then use startActivityForResult, not shared preferences

Reply:
okk

Reply:
> Then use startActivityForResult, not shared preferences

I don't think it can be done. Because the above situation(using preference manager) was done to handle if changes occur in another file and after that user presses back. Now, if we try to go for onActivityForResult when the user deletes the feedback on allReviewActivity page, aren't we forcing the user to go back(I mean we are performing the back action in code)?

Reply:
@iamareebjamal I tried implementing it. But the scene here is different from the normal cases. I don't think it is feasible to use startActivityForResult here.

Reply:
How is it different. You want to know if the child activity has deleted certain reviews. It's perfect